src/blocknative.rs

```diff
@@ -0,0 +1,129 @@
+use super::{GasPriceEstimating, Transport};
+use anyhow::{anyhow, Context, Result};
+use serde::Deserialize;
+use std::time::Duration;
+
+// Gas price estimation with https://www.blocknative.com/gas-estimator , api https://docs.blocknative.com/gas-platform#example-request .
+
+const API_URI: &str = "https://api.blocknative.com/gasprices/blockprices";
+
+pub struct BlockNative<T> {
+    transport: T,
+    api_key: String,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct EstimatedPrice {
+    confidence: u64,
+    price: f64,
+    max_priority_fee_per_gas: f64,
+    max_fee_per_gas: f64,
+}
```

Comment:
You say that this PR involves "legacy fee" but the terms I see here in this struct resemble post EIP-1559 lingo.

Reply:
From what I understood, this PR is using the EIP-1559 enabled gas estimates from BlockNative API, but returning them in "legacy fee format" because the `GasPriceEstimating` trait isn't EIP-1559 aware:

```rust
trait GasPriceEstimating {
     async fn estimate_with_limits(&self, _gas_limit: f64, _time_limit: Duration) -> Result<f64>;
}
```

Reply:
Nice! One small step for EIP-1559 support, on giant leap for improved gas estimation!